### ISSUE TITLE
Improve reliability of `Deploy Queues` jobs in PR builds

### ISSUE THREAD

**Issue by dougbu** [dotnet/dnceng#484]
PR builds in [`dotnet-helix-machines-ci`](https://dev.azure.com/dnceng/internal/_build?definitionId=596) frequently fail when builds of 'main' are running in parallel. This is especially true when the `Deploy Queues` job in a PR build overlaps `Build Images` in a main build. A PR builds may attempt to deploy a queue (create its scale set) using an image the main build is currently generating. This occurred w/ the ubuntu.2204.amd64 queue in [#20230731.03](https://dev.azure.com/dnceng/internal/_build/results?buildId=2232957&view=results) (a PR build) and [#20230731.01](https://dev.azure.com/dnceng/internal/_build/results?buildId=2232869&view=results) (a build of main).

Specific errors centre on the following in the Azure Resource Management template for new PR scale sets
``` json
"osImage": {
    "value": null
}
```
In particular
``` text
Deployment template validation failed: 'The value for the template parameter 'osImage' at line '75' and column '16' is not provided. Please see https://aka.ms/arm-create-parameter-file for usage details.'.
```

The root cause appears to be twofold:
1. We have no separate production and staging tables listing globally-available images. This makes sense because production builds and rollouts in particular should not create or deploy new images. The https://helixosautostaging.table.core.windows.net/generatedimages table hasn't been updated since 2021 for this reason.
2. The `Deploy Queues` logic ignores whether the `ImageId` column in the global images table is empty when choosing what to deploy. This part needs to be fixed.

One workaround would be extreme care to avoid `FORCE_QUEUE` for queues using images shown w/ `State=="GeneratingImage"` or `ImageId=""` in <https://helixosauto.table.core.windows.net/generatedimages>.

A more user-friendly option would be to ignore all but `Ready` images in `DeployQueues.Program.TryProcessDefinitionsAsync(...)`. Because PR deployment of images from the global table are usually of little interest, this seems safe. We can revisit if changes w/in the deployment code become common.

Chosen option is to reuse the `Deploy1ESHostedPools.DeploymentOperations.GetImagesToDeploy(...)` logic. This combines the above check for `Ready` images in PR builds with a failure when non-PR builds attempt to deploy an unready image. The second error is extremely unlikely but worth including, at least for consistency. In addition, the code we'll mimic focuses on the `ImageId` value in the images table. This is slightly more direct than using the `State` column.

Note searching back for less recent images to deploy would require a broader change to the `DeployQueues` code. Images are matched on their `DefinitionId` values and that's a hash of all artefacts and other important values in the definition. Using another image w/ the same name (not `DefinitionId`) makes PR-created scale sets less indicative of what later builds of main will or won't do.

### Release Note Category

- [x] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements

### Release Note Description

Ignore listed images that aren't ready when deploying Helix queues in PR builds. This situation now results just in an informational message.

In addition, introduced new errors if the latest listed images are not ready in non-PR builds. This case is extremely unlikely but covered for defence in depth and to improve alignment between our `Deploy1ESHostedPools` and `DeployQueues` code.

**Comment by dougbu** [comment]
completed w/ merge of [!32895](https://dev.azure.com/dnceng/internal/_git/dotnet-helix-machines/pullrequest/32895). waiting for rollout wasn't really necessary
